code/modules/client/preferences.dm

```diff
@@ -492,3 +492,130 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			default_randomization[preference_key] = RANDOM_ENABLED
 
 	return default_randomization
+
+/datum/preferences/proc/handle_client_importing(list/raw_data)
+	if(IsAdminAdvancedProcCall())
+		return
+
+	var/raw_data_version = raw_data["version"]
+	var/imported_data_version = isnum(raw_data_version) ? raw_data_version : text2num(raw_data_version)
+	if(imported_data_version < PREFERENCES_VERSION_MINIMUM)
+		to_chat(parent, span_userdanger("Cannot import this savefile, it is too old!"))
+
+	if(imported_data_version != PREFERENCES_VERSION_CURRENT)
+		if(tgui_alert(
+			parent,
+			"Importing a savefile which is older, this might cause issues.",
+			"Savefile Import",
+			list("Attempt Import", "Cancel"),
+		) != "Attempt Import")
+			return
+
+	var/static/regex/character_slot_regex = regex(@"^character[0-9]+$")
+	for(var/data_key in raw_data)
+		if(character_slot_regex.Find(data_key))
+			var/slot_id = text2num(replacetext(data_key, "character", ""))
+			if(slot_id <= max_save_slots)
+				continue
```

Comment:
this seems hacky and also if i understand things correctly, it will lock out people importing saves from when they had byond membership after they lose it.

Reply:
also i think the check is reversed

Reply:
it is indeed reversed
```suggestion
	var/static/regex/character_slot_regex = regex(@"^character[0-9]+$")
	for(var/data_key in raw_data)
		if(character_slot_regex.Find(data_key))
			var/slot_id = text2num(replacetext(data_key, "character", ""))
			if(slot_id > max_save_slots)
				continue
```